### DISCUSSION TITLE
__ANDROID_API__ redefined after the upgrade from R20b to R23b

### DISCUSSION THREAD

**Discussion by little-strong666** [android/ndk#1824] [Q&A] [answered]
The compilation macro -D__ANDROID_API__=28 is defined when r20b is used.
A compilation error is reported after the upgrade to R23b.
Variable __ANDROID_API__Redefinition
Does r23b's compiler define __ANDROID_API__?
And what is the value of __ANDROID_API__?

**Comment by enh-google** [comment, marked as answer]
typically you'd expect the _compiler_ to set `__ANDROID_API__` --- it's been doing that since 2016 (https://reviews.llvm.org/D26385). specifically your `-target aarch64-linux-android19` (or whatever) gets turned into [the equivalent of] `-D__ANDROID_API__=19` (or whatever).

where _that_ comes from will depend on what build system you're using.
